cpp/include/cudf/copying.hpp

```diff
@@ -173,5 +173,28 @@ std::vector<column_view> slice(column_view const& input,
 std::vector<column_view> split(column_view const& input,
                                std::vector<size_type> const& splits);
 
+/**
+ * @brief   Returns a new column, where each element is selected from either @p lhs or 
+ *          @p rhs based on the value of the corresponding element in @p boolean_mask
+ *
+ * Selects each element i in the output column from either @p rhs or @p lhs using the following rule:
+ *          output[i] = (boolean_mask[i]) ? lhs[i] : rhs[i]
+ *          
+ * @throws cudf::logic_error if lhs and rhs are not of the same type
+ * @throws cudf::logic_error if lhs and rhs are not of the same length
+ * @throws cudf::logic_error if lhs and rhs do not have the same nullability
```

Comment:
Why? The nullability of `lhs` should be independent of `rhs`'s nullability.

Reply:
Good point. If any one or both columns are nullable, the output should be nullable. Elements chosen from the non-nullable column should be marked valid.